From patchwork Wed Feb  9 21:55:45 2022
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Michael Riesch <michael.riesch@wolfvision.net>
X-Patchwork-Id: 12740944
Return-Path: 
 <linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by smtp.lore.kernel.org (Postfix) with ESMTPS id 1E746C433FE
	for <linux-rockchip@archiver.kernel.org>;
 Wed,  9 Feb 2022 21:56:29 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20210309; h=Sender:
	Content-Transfer-Encoding:Content-Type:List-Subscribe:List-Help:List-Post:
	List-Archive:List-Unsubscribe:List-Id:MIME-Version:References:In-Reply-To:
	Message-Id:Date:Subject:Cc:To:From:Reply-To:Content-ID:Content-Description:
	Resent-Date:Resent-From:Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:
	List-Owner; bh=XLWE7RBiBJZKLk4cSTz2oET1FH8oSG7HL7zoGxeDnKc=; b=fie4GPZSuj5e8O
	Pr36v+dl8c6xZpyY3jExfIsNxMJpAJBTWNfCkZ3JJCtYXr1jdRb9Syghb11/X1k02MuPzZM9lcLQm
	Z2Zc5wsTpz2tdKNoKYhK7tfWNf7K/3A7eDiyxfEPb3d38JKY+DoYncuHlMWWF2qQTeyhSWZekSiNw
	uja/3QXoBI5C5eFYnnXJivdyYJWSyrHS4mkTEIdVTm2+3Hgo9uDSD1y6fkuD1MJtOEIMx4/dV8fFw
	VyYNnFVIo2ujadYv8LDJnECDBqrxB+Ir6+XBo4cLBdav6xCbG7Af19jU5mtDLx6WuHm5Am7D5c9gj
	NH46NJecGmzkc81vVI1A==;
Received: from localhost ([::1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.94.2 #2 (Red Hat Linux))
	id 1nHuwp-001kV2-3h; Wed, 09 Feb 2022 21:56:23 +0000
Received: from mail-ve1eur02on0600.outbound.protection.outlook.com
 ([2a01:111:f400:fe06::600]
 helo=EUR02-VE1-obe.outbound.protection.outlook.com)
 by bombadil.infradead.org with esmtps (Exim 4.94.2 #2 (Red Hat Linux))
 id 1nHuwZ-001kMp-Us; Wed, 09 Feb 2022 21:56:09 +0000
ARC-Seal: i=1; a=rsa-sha256; s=arcselector9901; d=microsoft.com; cv=none;
 b=lNkItvvwtei+k30ntSUo3FEZ0d5Qip83ll32gP35DYSD5HbdXnB2ZqHv2q98ix+6p//JwSTQXbv+n2BnoLiLz93TMjgCXXxTJnOlTFQXns3YM2z075txIekqN8iTQASbLUzW3jiY61bMT+WVPzB3ZD5nSPrd6Mf0OLypnefD0yRprQoEiMHqmDq/O9IcxsZLKQHDdBQKnL5YlbNjy7H6JktSRL05wJJX9mUJ9rtG8D/aLi+RBypVUHedudwqeZVX5s4l59vjgsi4ePyXKNMO1ptX3d+helIX4KuWJLfoAGNl7ZsYQufulGoQ7yh0g8vzUUbNogV/XwWBTXRU35185w==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector9901;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-AntiSpam-MessageData-ChunkCount:X-MS-Exchange-AntiSpam-MessageData-0:X-MS-Exchange-AntiSpam-MessageData-1;
 bh=yLbgpsASm0DdWNyCqP9doSRhaeJEjX2WjAOp3VoZBmM=;
 b=M/k/yD7DGoFBPTdPFIwuKlyrhrU8b8YvZFRuOwwnJi0Yro0K69Kyh0cyB7/GW5tDxAuGeHCOWOASpOl04H1j9Ec9RExcGzdOQQsTmlZ/AZMYX23WVEOtYFi4YnIx4g1LS7Eo4owlyRaztBUO2d5qsur6zUvPY55HmLJhF5ujYXHl7ILX8+LRGIGN+72H3x3EVgDDp5v2J7vjHWA5aWqL6wrx1T988H/AgpN22BcO6W8nlgMJEe9UJ+hrtH8fo3YVLCFaM1CBVVNLMX/ybNlJQsDtxqNRMS2eXV9wJqKzyYHQ0ux71Awkt7GWJ5pSS383kFapAfabqrbvhvs+6xvJqw==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=none; dmarc=none;
 dkim=none; arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=wolfvision.net;
 s=selector2;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=yLbgpsASm0DdWNyCqP9doSRhaeJEjX2WjAOp3VoZBmM=;
 b=FoJ/nBgrt+PC+cdpKNLoKz8htuHVKaVEJTM2FJK++NWH7OjoWFgMWsC9lxaaiEFXUoiqRuOWotHM2+yEqAu6C+Dn7wR5fBZc8YVpIUn0Ox82VpoTPQXwIFE4qlhdHifZ+P7U/xxi8QukeY+Zsjqh5JE4hD/XBYtIKqd7j3IP9wQ=
Authentication-Results: dkim=none (message not signed)
 header.d=none;dmarc=none action=none header.from=wolfvision.net;
Received: from HE1PR0802MB2426.eurprd08.prod.outlook.com (2603:10a6:3:e1::23)
 by VI1PR08MB4575.eurprd08.prod.outlook.com (2603:10a6:803:eb::26)
 with Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.4951.17; Wed, 9 Feb
 2022 21:56:04 +0000
Received: from HE1PR0802MB2426.eurprd08.prod.outlook.com
 ([fe80::e14f:197:5001:4c56]) by HE1PR0802MB2426.eurprd08.prod.outlook.com
 ([fe80::e14f:197:5001:4c56%3]) with mapi id 15.20.4951.019; Wed, 9 Feb 2022
 21:56:04 +0000
From: Michael Riesch <michael.riesch@wolfvision.net>
To: dri-devel@lists.freedesktop.org, devicetree@vger.kernel.org,
 linux-kernel@vger.kernel.org, linux-arm-kernel@lists.infradead.org,
 linux-rockchip@lists.infradead.org
Cc: David Airlie <airlied@linux.ie>, Daniel Vetter <daniel@ffwll.ch>,
 Rob Herring <robh+dt@kernel.org>, Heiko Stuebner <heiko@sntech.de>,
 Peter Geis <pgwipeout@gmail.com>,
 Nicolas Frattaroli <frattaroli.nicolas@gmail.com>,
 Michael Riesch <michael.riesch@wolfvision.net>,
 Ezequiel Garcia <ezequiel@vanguardiasur.com.ar>,
 Alex Bee <knaerzche@gmail.com>, Liang Chen <cl@rock-chips.com>,
 Sascha Hauer <s.hauer@pengutronix.de>
Subject: [PATCH v6 1/5] dt-bindings: gpu: mali-bifrost: describe clocks for
 the rk356x gpu
Date: Wed,  9 Feb 2022 22:55:45 +0100
Message-Id: <20220209215549.94524-2-michael.riesch@wolfvision.net>
X-Mailer: git-send-email 2.30.2
In-Reply-To: <20220209215549.94524-1-michael.riesch@wolfvision.net>
References: <20220209215549.94524-1-michael.riesch@wolfvision.net>
X-ClientProxiedBy: ZR0P278CA0197.CHEP278.PROD.OUTLOOK.COM
 (2603:10a6:910:44::8) To HE1PR0802MB2426.eurprd08.prod.outlook.com
 (2603:10a6:3:e1::23)
MIME-Version: 1.0
X-MS-PublicTrafficType: Email
X-MS-Office365-Filtering-Correlation-Id: d407f3a2-f53a-4f44-1eab-08d9ec16f7e8
X-MS-TrafficTypeDiagnostic: VI1PR08MB4575:EE_
X-Microsoft-Antispam-PRVS: 
 <VI1PR08MB4575DC5FE8E33C739B23262EF22E9@VI1PR08MB4575.eurprd08.prod.outlook.com>
X-MS-Oob-TLC-OOBClassifiers: OLM:6430;
X-MS-Exchange-SenderADCheck: 1
X-MS-Exchange-AntiSpam-Relay: 0
X-Microsoft-Antispam: BCL:0;
X-Microsoft-Antispam-Message-Info: 
 5bmUesViaLms2bDWroGduFry/PsD/b9GWJgffCzKfBqVk9rmnJUs6VOtORWR/2uSKdlZZ/cllkobm62Q21HXb28I3mXWhyyFn4DrGN5X3aGHUWzbwVRdEwA/Ghm8UdWwKrsXxZsNxYaRpsHHc0Vnqrjure07UlWWdVcmCJxtlSuskdz7ho8cnTLli+Z26BncUSfma2Cw1jHC/PSYUl6Z8dUksISqvVORAQ9JqXWvc66Z7I5LiMnTXXUZxoreutk5ZDKl4Fa6mwQ+PGg2d5A7dG69xttFSmu0tp6dhgZAkzYqenmQa5tunUHyp8qv+mh4zvSN4XgTWTdwzH/E4LBPJCSiGvyyoMl/a0uHgmiebr6dPAsERTvPRUc/yINciqHWLEdRddi/LKKNUFWLezQcTz3O0/GHelOF4fdbyc4TXIxaShU58huJtaAcVXHDQJsK5/SnZYcdCAu9RUshNdtuIvGJuR6dRJAWHKVmX/2Mm8KWnA/KrPycSMktEND3VnqTb7MJt3qeoUUAUZ+P4+MgzkJU+v8wbq/XXa0vTfH3vZdktbvCHqclR8tbEPmymQaXSFZIWCp1hXUKuSddjuZm5HOCx2ox0VJkGdHGWH+Kj87z9DfyS7wzZVQ5+s0jGic4DPzpoHlBnty4vKMocwPSIQ==
X-Forefront-Antispam-Report: CIP:255.255.255.255; CTRY:; LANG:en; SCL:1; SRV:;
 IPV:NLI; SFV:NSPM; H:HE1PR0802MB2426.eurprd08.prod.outlook.com; PTR:;
 CAT:NONE;
 SFS:(13230001)(4636009)(366004)(6512007)(38100700002)(2906002)(4326008)(508600001)(316002)(54906003)(52116002)(36756003)(6666004)(6506007)(6486002)(2616005)(44832011)(1076003)(86362001)(7416002)(66946007)(66556008)(186003)(8676002)(66476007)(8936002)(5660300002)(83380400001);
 DIR:OUT; SFP:1101;
X-MS-Exchange-AntiSpam-MessageData-ChunkCount: 1
X-MS-Exchange-AntiSpam-MessageData-0: 
 wGg90dOQBK5e7FWrPnNl0ukd+/SBrdomWsYJ+ix3xg6NqYn06r96gpppVbCDdOmFmnIaj+b25IwR1gEz+A+w30iu1O6Mgo8ciwx5RIkulycq57frxAuL9/YtOJZcgvOzTIBOYjBbZPITMXBpH8+BVDCMYeAwLqsO0sm/9YoGqfO12pUQ0rUPyVTqtBviciooPhQywwCiXsbcx1aaQ4DY5Qv18VK4vsWt4sVu3cr9TDFindRmxsQMv+oAU7f1yPipwVq/0UYvYiLxqZ0clwTb0SXTqRCPwF4jP4qfucgNmO7ujP5dV33P6fRComoy/gr1jTrcf/TkkScvtkXKpLrDYvw4/S//FbxZN+v9J6UrBO4FeI8yvAex0YyFTaJ/cfP8riOhwnxTjtmFaDzeUcC2z+QQoD+ALYxeEmoq1jJlcYKftwjGBI6j+khdj/YbC99LecFGPKjPzCesjgGZ+yqZ9/cBMlKevMf5koDibOFcM3JqoB8Tj4TT4rLsB4UGCUGJw0dfx/cSIdNoXvXlNEq1d14FPzb3RSBZjQLvCWxv16VxDDLzyg521k9oWFmWnT9Jk5ZGR5xJ4W9qNhI/sNXtgfhXFr4JiFjZqBheh07Lz1NxqdDkvZ0ZdeaxGIeAfWEQtOCobKvYtSN9njrXsUOfAcJORl7ibxMGxAhwirnTra56pM9la67wmQaBVp8B3I4Vq1UMEHpeDWqCG7PaoeS3S/V5VziMXIB8MmqyHisvYOYgqQFkYSUT2AZa6IIb6D4ABLrmx1CdrH68dQZ+Gy/WuxKnPrqlgZJKnxmBjdZ5SfOJZJzkQ+c1Ll8VdsOAmRJYWUDtguTuetVCXOJJCfswe8lsKMKfnMvWqTTMGYj1N0PbMRWyCh1FgnLJRTZRhvw5CCd1APp/Bg9OTbYh4dHTsTn7g7sdgYr9+B6BQ4xGfd06raaI3zDS6NeiI9D5i+vI+snFxhNnI51iY2uquDoXj30ZbOdOggIXjtesSzAn4R6ld+LQBh46T6hf1vE97QZZlWAXYhX5WjfeNu3taDK+SCNd5tG3HWaC/zjz0VZCVwB23KrA4IXMvd0+I0NTTuKwiVax3aNnMVeT76ffxrZU8UAJKwoG7eevPy387dVaIRG6F8HxXE6EwrpR3ugeCUVoA2McsEOk/QC0Ran9oiBfWq9QICOwvjqQMshXcZzIsqcNQ+zd8fvsDEBgDvATuprwtsbiHIJuO8eIKf6P8Rsd9RlRM2FdEV1gIrytUI9l6msHIR4C4gIY08rKclL+QU2z/zRWfv4eZbCH0BRpj7X3UtBuXxpntUhAqGZELw7P/6eba1X11mJ7fgcDfZKYJYk3Lms9H1oTzxU30kO66MbB4xHlgx12ErFLmKOGQgal/5kXZOAQjWWuP8LWFsexZw/bd5HqKE+AKuxuDdh/++Hzg6RA7VY11j3ucFbkqSEIrQKSJC5DULPTjSk36iHeroOXu4qkg58KhiADOZ5PWbVZwCK7AxH4fBW8oFtwdi4c10RCzyUJJc6X93CTR6EVf05bd1FMxrn2tjptPpeiCmZANNiWonEs30gkn1e9cHZSunv/KsmB5sycrUnmZTWHE8cS65ICFHsAxcbkKqi+LKvid4vnIf02JEMyJASt7emKy6+iSuwHtF7bsY9eo1p4WNxfswFsyHyoVRFkxLTaItvdZoPBSGdxT8khlvUY/v+07Bn1jNZCwkYcVCiqHNygh6qTG/uFC6n/aCofc+kJewA23+qKY7HmjfIxF5F+lkXuInc=
X-OriginatorOrg: wolfvision.net
X-MS-Exchange-CrossTenant-Network-Message-Id: 
 d407f3a2-f53a-4f44-1eab-08d9ec16f7e8
X-MS-Exchange-CrossTenant-AuthSource: 
 HE1PR0802MB2426.eurprd08.prod.outlook.com
X-MS-Exchange-CrossTenant-AuthAs: Internal
X-MS-Exchange-CrossTenant-OriginalArrivalTime: 09 Feb 2022 21:56:04.4287 (UTC)
X-MS-Exchange-CrossTenant-FromEntityHeader: Hosted
X-MS-Exchange-CrossTenant-Id: e94ec9da-9183-471e-83b3-51baa8eb804f
X-MS-Exchange-CrossTenant-MailboxType: HOSTED
X-MS-Exchange-CrossTenant-UserPrincipalName: 
 stQ/ariryfiFcXHKPRMprDzk9VAFL6a6cH/YmlKsGRiFaQ+NxvJsNJrBa4cEGh5Bi+lIwMjj9EImOzbUagNp4T5XIqIRFYFG8bO5lG5g390=
X-MS-Exchange-Transport-CrossTenantHeadersStamped: VI1PR08MB4575
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20220209_135608_077044_387FCBEF 
X-CRM114-Status: UNSURE (   9.96  )
X-CRM114-Notice: Please train this message.
X-BeenThere: linux-rockchip@lists.infradead.org
X-Mailman-Version: 2.1.34
Precedence: list
List-Id: Upstream kernel work for Rockchip platforms
 <linux-rockchip.lists.infradead.org>
List-Unsubscribe: <http://lists.infradead.org/mailman/options/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-rockchip/>
List-Post: <mailto:linux-rockchip@lists.infradead.org>
List-Help: <mailto:linux-rockchip-request@lists.infradead.org?subject=help>
List-Subscribe: <http://lists.infradead.org/mailman/listinfo/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=subscribe>
Sender: "Linux-rockchip" <linux-rockchip-bounces@lists.infradead.org>
Errors-To: 
 linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org

From: Alex Bee <knaerzche@gmail.com>

The Bifrost GPU in Rockchip RK356x SoCs has a core and a bus clock.
Reflect this in the SoC specific part of the binding.

Signed-off-by: Alex Bee <knaerzche@gmail.com>
[move the changes to the SoC section]
Signed-off-by: Michael Riesch <michael.riesch@wolfvision.net>
---
 .../devicetree/bindings/gpu/arm,mali-bifrost.yaml | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/Documentation/devicetree/bindings/gpu/arm,mali-bifrost.yaml b/Documentation/devicetree/bindings/gpu/arm,mali-bifrost.yaml
index 63a08f3f321d..4d6bfae0653c 100644
--- a/Documentation/devicetree/bindings/gpu/arm,mali-bifrost.yaml
+++ b/Documentation/devicetree/bindings/gpu/arm,mali-bifrost.yaml
@@ -159,6 +159,21 @@ allOf:
         power-domains:
           maxItems: 1
         sram-supply: false
+  - if:
+      properties:
+        compatible:
+          contains:
+            const: rockchip,rk3568-mali
+    then:
+      properties:
+        clocks:
+          minItems: 2
+        clock-names:
+          items:
+            - const: gpu
+            - const: bus
+      required:
+        - clock-names
 
 examples:
   - |

From patchwork Wed Feb  9 21:55:46 2022
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Michael Riesch <michael.riesch@wolfvision.net>
X-Patchwork-Id: 12740946
Return-Path: 
 <linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by smtp.lore.kernel.org (Postfix) with ESMTPS id 81014C433F5
	for <linux-rockchip@archiver.kernel.org>;
 Wed,  9 Feb 2022 21:57:02 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20210309; h=Sender:
	Content-Transfer-Encoding:Content-Type:List-Subscribe:List-Help:List-Post:
	List-Archive:List-Unsubscribe:List-Id:MIME-Version:References:In-Reply-To:
	Message-Id:Date:Subject:Cc:To:From:Reply-To:Content-ID:Content-Description:
	Resent-Date:Resent-From:Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:
	List-Owner; bh=NukViYEImO7Q5YYiC57fic2/c6oMpdSoVR6o1GHfSRU=; b=wJhH/bkhqWLGnB
	hc5fYgM6gslyhaegrSlyB1Z76DeoCj5zyQlOlNr3boJZA6pFkFA+nyNWSsIBvNJN3b1CV48Wl+ZO4
	0SzhBnJLoz/cWuoY/FExgL4cC8evkMEIQ9ZdtEVWFLdrH1wiruFkgbAYxHIMdH8kb6O3EhT+Ot4HA
	zep7+mPUfofLk5ubZu/9/cqklczjGknbF1rQPQ7uMQ2RUgbME5slKN7v9CQEEIDRZYHpHzZv6wGzO
	tbx0gCEwQdA3AuZLWLAUpa+KipHMmO4QHquZ62KCQcAR2dnOQjo2KSGZgrJqb9KyiY0qjTyfXCO7I
	vaKNbk3lXjLtSIhyhNhg==;
Received: from localhost ([::1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.94.2 #2 (Red Hat Linux))
	id 1nHuxN-001koQ-4t; Wed, 09 Feb 2022 21:56:57 +0000
Received: from mail-ve1eur02on0600.outbound.protection.outlook.com
 ([2a01:111:f400:fe06::600]
 helo=EUR02-VE1-obe.outbound.protection.outlook.com)
 by bombadil.infradead.org with esmtps (Exim 4.94.2 #2 (Red Hat Linux))
 id 1nHuwd-001kMp-Gp; Wed, 09 Feb 2022 21:56:13 +0000
ARC-Seal: i=1; a=rsa-sha256; s=arcselector9901; d=microsoft.com; cv=none;
 b=SZwKFZAvIcH/usRBsdnJr354EbH0diPRn18hMxE8hmjYeKrcP/k7Hl4a2tvfrBIoB+4Q3Sy6TUDE44Qr2OwtLeR+deUL/JOVWyf9XC8fGG8XvaxY861xZca6WSbxzySVxFXG23rKxjstak05BfGgyWxovJ1Bi/F8y2t5GzaWus4ezyOj4saL+JswT77A8nsVBD7zBzmthnguYTuq5Bw+cPewRA6gktOqCKPfQhtGUv4I43eGDGoGzspjbdtgHrP9r5NQLJoCXmCzpByfrWCWkfV0DH2tVYleFfTj++8CRbRSgSbdhJC300aOAlt7iFkmHDe8HcMSKdlNhqyIJv41sw==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector9901;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-AntiSpam-MessageData-ChunkCount:X-MS-Exchange-AntiSpam-MessageData-0:X-MS-Exchange-AntiSpam-MessageData-1;
 bh=6HEqSY99564YBUXrObStmi4N7YBFLburRTVPEPTCo3E=;
 b=N4GZ1fCck68ahleohoNjrcU5imPsevGWYZ7tWPZL7i4KXu/qvWOAlmjDJn7NPStvOAM1+tk1h6nnzt05CcjSqGV0zfe0V8DoQRDJOyzaSfZUTx/9BGY/sQLvazNCF90iUdvvLyAeNznkUPRgRHSFT4LtX1zvDlHZbtlYB6Y+llNspeOaaOvWJWqsfWmeZKmZwnteYbDKkE8Htl2PuaM4O80mbjyZDGOvobsoYlW3nI6f14UMgPCCmZJzSAz+7kmE2X1V4pd09468teK3PLDugz/OFX686Qeprma9SmSCiW8CzP+rr5Y21Wtu2/tfiyF73/2P13zpsyYXEWpEGSD4vA==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=none; dmarc=none;
 dkim=none; arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=wolfvision.net;
 s=selector2;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=6HEqSY99564YBUXrObStmi4N7YBFLburRTVPEPTCo3E=;
 b=uslJbxBekQ0+iu+dxg0wvqieQPoijoEd7dGOzp+jpCaCYEAGIE2fRYNBSZIZKJT4jlRlcv0xXMuRBGRCqCp7RPTvOQVXDhyMg1UIs2pUDSvj+qHKiHLPULpx+1r3gegmA8iApuBWkjw64mNlvnXbjxAqFf5cNvH3cSxQNR+BeFM=
Authentication-Results: dkim=none (message not signed)
 header.d=none;dmarc=none action=none header.from=wolfvision.net;
Received: from HE1PR0802MB2426.eurprd08.prod.outlook.com (2603:10a6:3:e1::23)
 by VI1PR08MB4575.eurprd08.prod.outlook.com (2603:10a6:803:eb::26)
 with Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.4951.17; Wed, 9 Feb
 2022 21:56:06 +0000
Received: from HE1PR0802MB2426.eurprd08.prod.outlook.com
 ([fe80::e14f:197:5001:4c56]) by HE1PR0802MB2426.eurprd08.prod.outlook.com
 ([fe80::e14f:197:5001:4c56%3]) with mapi id 15.20.4951.019; Wed, 9 Feb 2022
 21:56:06 +0000
From: Michael Riesch <michael.riesch@wolfvision.net>
To: dri-devel@lists.freedesktop.org, devicetree@vger.kernel.org,
 linux-kernel@vger.kernel.org, linux-arm-kernel@lists.infradead.org,
 linux-rockchip@lists.infradead.org
Cc: David Airlie <airlied@linux.ie>, Daniel Vetter <daniel@ffwll.ch>,
 Rob Herring <robh+dt@kernel.org>, Heiko Stuebner <heiko@sntech.de>,
 Peter Geis <pgwipeout@gmail.com>,
 Nicolas Frattaroli <frattaroli.nicolas@gmail.com>,
 Michael Riesch <michael.riesch@wolfvision.net>,
 Ezequiel Garcia <ezequiel@vanguardiasur.com.ar>,
 Alex Bee <knaerzche@gmail.com>, Liang Chen <cl@rock-chips.com>,
 Sascha Hauer <s.hauer@pengutronix.de>,
 Ezequiel Garcia <ezequiel@collabora.com>
Subject: [PATCH v6 2/5] arm64: dts: rockchip: add gpu node to rk356x
Date: Wed,  9 Feb 2022 22:55:46 +0100
Message-Id: <20220209215549.94524-3-michael.riesch@wolfvision.net>
X-Mailer: git-send-email 2.30.2
In-Reply-To: <20220209215549.94524-1-michael.riesch@wolfvision.net>
References: <20220209215549.94524-1-michael.riesch@wolfvision.net>
X-ClientProxiedBy: ZR0P278CA0197.CHEP278.PROD.OUTLOOK.COM
 (2603:10a6:910:44::8) To HE1PR0802MB2426.eurprd08.prod.outlook.com
 (2603:10a6:3:e1::23)
MIME-Version: 1.0
X-MS-PublicTrafficType: Email
X-MS-Office365-Filtering-Correlation-Id: cc481682-3152-45a6-a9f9-08d9ec16f8fc
X-MS-TrafficTypeDiagnostic: VI1PR08MB4575:EE_
X-Microsoft-Antispam-PRVS: 
 <VI1PR08MB4575DC23B3EAA88AC87536B8F22E9@VI1PR08MB4575.eurprd08.prod.outlook.com>
X-MS-Oob-TLC-OOBClassifiers: OLM:983;
X-MS-Exchange-SenderADCheck: 1
X-MS-Exchange-AntiSpam-Relay: 0
X-Microsoft-Antispam: BCL:0;
X-Microsoft-Antispam-Message-Info: 
 aK6qfnw4RikqbqpmRQ/CMr70Mxlf6TIU+NWU+cA+BHm4LKNZ7kL94Ic5y8KkGauB7EiQJO6GbCULoAqTzXf4zPMDCwPulpXE4Q7Vxm9J6FzFu31AfwtUP0jo/IaKX7ankNOPk79xw28iDSxw8kM7pW7jamQd9PJanHyVlUsy8PYxSc9OpCPV0033hcF3wYugHdqO1o97BDgQ3jxKz6fVlTEAIHP7SzHGYwc9DSyMZGFoZTsp9tMN2bkxDOh1cOoeCgANI7wXRaiPYnChW80EblkqOB34Hg9EVxzO/7a62XrEtiIiqcTYWdP/O9gm/mieIutPyLGlJSHdcpkAAiI3YBRtbzqOlPr8WFMfVRpVGGnrlZFPO3KsEU2569GFfiKpjtzyuf6dNdbhnTa2LOKXrBP4bsoF+Hae3AJVsFFHaT8hWaPs3maof3hf/T0/13y7QKkUfYRm0pkiNuoNly6pG4/XYbpzRmoEBs7HEIh2mxWhTOV8YiJWEFBm5S2ep5gbDsprN/5KudYMvwq/QzOp2mJmoc+se7X5qtQOVbOGU1Aea4HQ8s0GauUBK6iba9twxJ35DSmFhmTa7zWUa6IF5ARrxS7Zt8uu7L/pDXIS5FreIoTaKYyXVNUvI2RvRacw+4neyRP5HCgbOiqwTey12A==
X-Forefront-Antispam-Report: CIP:255.255.255.255; CTRY:; LANG:en; SCL:1; SRV:;
 IPV:NLI; SFV:NSPM; H:HE1PR0802MB2426.eurprd08.prod.outlook.com; PTR:;
 CAT:NONE;
 SFS:(13230001)(4636009)(366004)(6512007)(38100700002)(2906002)(4326008)(508600001)(316002)(54906003)(52116002)(36756003)(6666004)(6506007)(6486002)(2616005)(44832011)(1076003)(86362001)(7416002)(66946007)(66556008)(186003)(8676002)(66476007)(8936002)(5660300002)(83380400001);
 DIR:OUT; SFP:1101;
X-MS-Exchange-AntiSpam-MessageData-ChunkCount: 1
X-MS-Exchange-AntiSpam-MessageData-0: 
 poajkVAnLCDqHOvV1rjvOGY5gxxPNBH5TD2wtrp1kTxDcaU15YOALQpoKPGFcsr/a62QJ/MHi93UNx+H4OOhLCAPFTcyTHFNCo+61A8e5U3AisriKZAjjXF8agYMLy/0wIbqBaas3jDhAmL1GvYRmdeLDdHKX0aPN5sCUxYQ3T9Ei+hkDDCyl7rbY2vOT+gda01Yf/QZHjglXfvdI/ckyMZl31ILzYaytOL8BjpCG/fTbjwT/EKNzgwiM5W8mHcekEFA7iwWg6j218wLIodLZ7bHIcweIovAtv9tdD20MteIIkuDUZ3eoMSGghVDt/Xf+4g5opS8oMAPr8VE8FW+kIkWeWfVXJtOAOXnf5juFchDuUpOXQ0cifMsTDBNMyIPsIcf6GdVhAmjDRY6LfdzGSFJh2yI9UJHVztbAKdYp12fCihaU7OBHR/IfluGOmScOGSQVaKdMmy28KJxBdWPV9FqudoYG3tNdvvtAYsGldmzA4Af6PsFgskGqaERuI4DuE8HsxGJtVUvxnq4i8yK6sCh47S+I+hczaP+foQfoTE2ZQ3lXzn1nQvoWoqCiRcp6Xq+rNGvhHvLLvIHG9dfZGXqSLugYK2X5deCZzzS2uyQiJfANkSt8UL+34LsJCZKrXvOevIpApGISAbiknvj6/rOJ9fipBAgYblOAacEJ1+c6K9dwWtEkYRG8kIsO5n3KqVqRKZeRw7N4ki8jSER+YXGii0eL/3+6U0izeQLOw+ggIrTBaARnqFXwP/1Fkyu5LEClZufBAqZU4uECYlGx8EeptQFqjbM7syL/00McpYmgc64i4HlooigF9KiwNMNWRtiGbEZZey8YszPraBvGzN823Rt7HwlrzCDR1nMWsUtNGj+yqDMtw677FdDHJ93nlfsvVJXk9FDeEtUSRZWvvjhxUv5ySyJ2Z9FC21HZ3KVCePlPsZZCqh+cvnV+pZ4cKACl4fi8Tj7TR0xLRG4QxrGGk1/JRXNf/92wLXU+ZHVgg/5AA7nmdpeziocdq3eBwJyKPgwcB7f+U3rQlBfHrkd2F8j7NeqpuB7uTb1FedrDELpaki9IcMeWs3gMiCkohTdb+O0vN4lHsf0crdsBIDKT7Vq4Qp+tXx8p0pK7eY0GjbwKUSD1bvevyrlgfeRDhLRMmrHYkPxpJ9hQVJYjvxC2m1QrDsO68EaO87hEPafZKjpD3xej6I7rfOKFQ2dmRs2R9tIghhXuAf6VK2oyCDxEnayfmSBTyutQgv2pBbVvBQw9wWPhT3lGo3nsDiT4LVIshfomZ8W2Eglb8dD6gDPwZEfMEa7DX4RK9J8iF1oAzQmNIB0MKJv8PBan9Cta2OQKs2694OXqCU+52h/fJRGwYZVaa8Xf19fAuy/gayOxdhqT2Xq+PrgYdPXxE4+Qlza+6Sstp6iUIua9e16He+btlGIjwSO3Iy8xAmxFFZNu182UNRG0WmlHfuEaySugwiPeRO4P9CnPuPI4WXP8+Lr7hkAxkyQlW2ozL3dE1k7qvrq3y558iDBD8M6XnDXDv//ppP5kVqBpcVa0JbujaPabixoDcppWkLTaK+PBljYa/TumN3wtPicznYPY9g/aOEWhQGCUrJHU8Pi/jeJ5A/JaMAAUwlw9E6E7sPYgAHc4pu9ORTCDM2p+Ill08aXU974p99ACD/YM+qDC1jQ3Rztp5SyGD+n4NAEyV9c2PzeEY8IYydG9BWTP6YXGB7UP6qFhHgJ1LwqdSrgQ5LrbHrJ63tZB/k9GHawvYSqtp8=
X-OriginatorOrg: wolfvision.net
X-MS-Exchange-CrossTenant-Network-Message-Id: 
 cc481682-3152-45a6-a9f9-08d9ec16f8fc
X-MS-Exchange-CrossTenant-AuthSource: 
 HE1PR0802MB2426.eurprd08.prod.outlook.com
X-MS-Exchange-CrossTenant-AuthAs: Internal
X-MS-Exchange-CrossTenant-OriginalArrivalTime: 09 Feb 2022 21:56:06.1629 (UTC)
X-MS-Exchange-CrossTenant-FromEntityHeader: Hosted
X-MS-Exchange-CrossTenant-Id: e94ec9da-9183-471e-83b3-51baa8eb804f
X-MS-Exchange-CrossTenant-MailboxType: HOSTED
X-MS-Exchange-CrossTenant-UserPrincipalName: 
 +5w+oLRr84eksY5rZsRTyuRwk8mlXkXMFSpNx2TV+Lweir+yyRNNE3+AtgpuWUVHJ2juAx3ZiPdn2ltRCbXnDoiuVgO/3oz82bYracmqlK8=
X-MS-Exchange-Transport-CrossTenantHeadersStamped: VI1PR08MB4575
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20220209_135611_615157_529C700A 
X-CRM114-Status: GOOD (  10.72  )
X-BeenThere: linux-rockchip@lists.infradead.org
X-Mailman-Version: 2.1.34
Precedence: list
List-Id: Upstream kernel work for Rockchip platforms
 <linux-rockchip.lists.infradead.org>
List-Unsubscribe: <http://lists.infradead.org/mailman/options/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-rockchip/>
List-Post: <mailto:linux-rockchip@lists.infradead.org>
List-Help: <mailto:linux-rockchip-request@lists.infradead.org?subject=help>
List-Subscribe: <http://lists.infradead.org/mailman/listinfo/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=subscribe>
Sender: "Linux-rockchip" <linux-rockchip-bounces@lists.infradead.org>
Errors-To: 
 linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org

From: Ezequiel Garcia <ezequiel@collabora.com>

Rockchip SoCs RK3566 and RK3568 have a Mali Gondul core
which is based on the Bifrost architecture. It has
one shader core and two execution engines.

Quoting the datasheet:

Mali-G52 1-Core-2EE
* Support 1600Mpix/s fill rate when 800MHz clock frequency
* Support 38.4GLOPs when 800MHz clock frequency

Signed-off-by: Ezequiel Garcia <ezequiel@collabora.com>
Signed-off-by: Alex Bee <knaerzche@gmail.com>
Signed-off-by: Michael Riesch <michael.riesch@wolfvision.net>
---
 arch/arm64/boot/dts/rockchip/rk356x.dtsi | 49 ++++++++++++++++++++++++
 1 file changed, 49 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk356x.dtsi b/arch/arm64/boot/dts/rockchip/rk356x.dtsi
index ff1689283996..50bbea862a6a 100644
--- a/arch/arm64/boot/dts/rockchip/rk356x.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk356x.dtsi
@@ -144,6 +144,40 @@ scmi_clk: protocol@14 {
 		};
 	};
 
+	gpu_opp_table: opp-table-1 {
+		compatible = "operating-points-v2";
+
+		opp-200000000 {
+			opp-hz = /bits/ 64 <200000000>;
+			opp-microvolt = <825000>;
+		};
+
+		opp-300000000 {
+			opp-hz = /bits/ 64 <300000000>;
+			opp-microvolt = <825000>;
+		};
+
+		opp-400000000 {
+			opp-hz = /bits/ 64 <400000000>;
+			opp-microvolt = <825000>;
+		};
+
+		opp-600000000 {
+			opp-hz = /bits/ 64 <600000000>;
+			opp-microvolt = <825000>;
+		};
+
+		opp-700000000 {
+			opp-hz = /bits/ 64 <700000000>;
+			opp-microvolt = <900000>;
+		};
+
+		opp-800000000 {
+			opp-hz = /bits/ 64 <800000000>;
+			opp-microvolt = <1000000>;
+		};
+	};
+
 	pmu {
 		compatible = "arm,cortex-a55-pmu";
 		interrupts = <GIC_SPI 228 IRQ_TYPE_LEVEL_HIGH>,
@@ -444,6 +478,21 @@ power-domain@RK3568_PD_RKVENC {
 		};
 	};
 
+	gpu: gpu@fde60000 {
+		compatible = "rockchip,rk3568-mali", "arm,mali-bifrost";
+		reg = <0x0 0xfde60000 0x0 0x4000>;
+		interrupts = <GIC_SPI 40 IRQ_TYPE_LEVEL_HIGH>,
+			     <GIC_SPI 41 IRQ_TYPE_LEVEL_HIGH>,
+			     <GIC_SPI 39 IRQ_TYPE_LEVEL_HIGH>;
+		interrupt-names = "job", "mmu", "gpu";
+		clocks = <&scmi_clk 1>, <&cru CLK_GPU>;
+		clock-names = "gpu", "bus";
+		#cooling-cells = <2>;
+		operating-points-v2 = <&gpu_opp_table>;
+		power-domains = <&power RK3568_PD_GPU>;
+		status = "disabled";
+	};
+
 	sdmmc2: mmc@fe000000 {
 		compatible = "rockchip,rk3568-dw-mshc", "rockchip,rk3288-dw-mshc";
 		reg = <0x0 0xfe000000 0x0 0x4000>;

From patchwork Wed Feb  9 21:55:47 2022
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Michael Riesch <michael.riesch@wolfvision.net>
X-Patchwork-Id: 12740947
Return-Path: 
 <linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by smtp.lore.kernel.org (Postfix) with ESMTPS id BD882C433FE
	for <linux-rockchip@archiver.kernel.org>;
 Wed,  9 Feb 2022 21:57:28 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20210309; h=Sender:
	Content-Transfer-Encoding:Content-Type:List-Subscribe:List-Help:List-Post:
	List-Archive:List-Unsubscribe:List-Id:MIME-Version:References:In-Reply-To:
	Message-Id:Date:Subject:Cc:To:From:Reply-To:Content-ID:Content-Description:
	Resent-Date:Resent-From:Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:
	List-Owner; bh=Jo8Uf5/WB6xxt5kGCBvD1qwNKIbejWx5dhQaWNLcGK4=; b=vYoIBU5avTsV/Y
	kv5Vz0XZblBJnOb5Lvb1IfJQwgfLpnykmE/alj46B6QMdChXbN5ZmC2el1ymH7tXYZqQsNHodPo2o
	yngP9BXKex8KmyEJ7pP3heaZCT72xIDYEJohFEtSGJAkEP9f2QEVNZnzU8bUxXyeHST7WnDXPvG/8
	kF54BDvmibNWIGc2Cpq2TY4E7YN7g+xyyNrSVBh2egsMhJB9/bF1Ra2Hiz4C/g3hFo6NvnfIVbWeK
	k5vCyKQ2kcvkWbSSvXcihkJ2pTvr4zv9IXUZHJZAyUUr0S3aB4SUKLIjPkjUhOX0t/nYRZmA11smH
	0aB1gJM7tQhFJNt2erAg==;
Received: from localhost ([::1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.94.2 #2 (Red Hat Linux))
	id 1nHuxo-001l1r-6m; Wed, 09 Feb 2022 21:57:24 +0000
Received: from mail-ve1eur02on0600.outbound.protection.outlook.com
 ([2a01:111:f400:fe06::600]
 helo=EUR02-VE1-obe.outbound.protection.outlook.com)
 by bombadil.infradead.org with esmtps (Exim 4.94.2 #2 (Red Hat Linux))
 id 1nHuwf-001kMp-LS; Wed, 09 Feb 2022 21:56:15 +0000
ARC-Seal: i=1; a=rsa-sha256; s=arcselector9901; d=microsoft.com; cv=none;
 b=bciXlOzgDOeLXgf5zKgAWDfSo4HAigJolDRGpsar3JEZYIyBlkaHTWIYkv6aDCWt67bYdN3WwgajG1SbnYvTsHV89WlOJ8gFq1YEKjlhwcY1qFk1pY0j0fPgkZXTr47snafrkGcsUbakxbsQ6ISv1GlCtjdXldxX3/QyXrxjMIXonl5YS6TwKS5XgiXdAaHHQaZDbiBAeEzllEChh3xYezYhzVjbp5UflCsoVh0/PqP41w7JP5XupJ0e44DcxRMb9X1AqifcUtkR2pi8EOlVCbM+RaOhhNpZ+0G6n6cWo+3zCKwN5MGH85O476wpwtqwEq4l2ePeJ3rHt49dduTCjA==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector9901;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-AntiSpam-MessageData-ChunkCount:X-MS-Exchange-AntiSpam-MessageData-0:X-MS-Exchange-AntiSpam-MessageData-1;
 bh=4gcqtk8JERjBYgVYJAmds3gb0AFb3R3KmOp9CD1Uczo=;
 b=ZDxq1xgrffa8+29EvpvUiS36Db5k+yBZ/9PCaahHWpDyPWnUFxML7itRwWwdnl2VLiYV4GHA7sw+Wa3nZvsj0GU/HhtG1W20nWjt6a5CD/SQ6yNufDgJwfbQqLh4HeL5l0ks8MGTGATOgtK2j1hhMJEzo7L84Kf4HHA+lh0tvZ5HdCIoAXiR1oZJQAhl6T3pKdfmL92QGs8Xx7tvQ/48nM7ikVDa+yuheo7bXZkLoPdrx+0Mrcz3FulOjuZNa20xhrBo3nQaUl1q1q0BPznFYJKWbkz+m8m9FnhE7hk68XGvSHGCKLrbCyMscIbO5XBwaHTJj1NeOzSrcktChv5ivw==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=none; dmarc=none;
 dkim=none; arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=wolfvision.net;
 s=selector2;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=4gcqtk8JERjBYgVYJAmds3gb0AFb3R3KmOp9CD1Uczo=;
 b=MNouepg80qIRU/ImwH+6siCCa4z7LhZpg6MVME9soHF8lBaep+HkHKIVRDCSRwHOGEhUIXddVxdvzFoc9dNXKtSRZY+iyUYSt1YyTi+tn3LfEpRbKeHZHqfLM2lvaTSNYwZpt7cQD5iw5bEz/hMLkjUHhI+XtLhNzlBPD7uSOb8=
Authentication-Results: dkim=none (message not signed)
 header.d=none;dmarc=none action=none header.from=wolfvision.net;
Received: from HE1PR0802MB2426.eurprd08.prod.outlook.com (2603:10a6:3:e1::23)
 by VI1PR08MB4575.eurprd08.prod.outlook.com (2603:10a6:803:eb::26)
 with Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.4951.17; Wed, 9 Feb
 2022 21:56:08 +0000
Received: from HE1PR0802MB2426.eurprd08.prod.outlook.com
 ([fe80::e14f:197:5001:4c56]) by HE1PR0802MB2426.eurprd08.prod.outlook.com
 ([fe80::e14f:197:5001:4c56%3]) with mapi id 15.20.4951.019; Wed, 9 Feb 2022
 21:56:08 +0000
From: Michael Riesch <michael.riesch@wolfvision.net>
To: dri-devel@lists.freedesktop.org, devicetree@vger.kernel.org,
 linux-kernel@vger.kernel.org, linux-arm-kernel@lists.infradead.org,
 linux-rockchip@lists.infradead.org
Cc: David Airlie <airlied@linux.ie>, Daniel Vetter <daniel@ffwll.ch>,
 Rob Herring <robh+dt@kernel.org>, Heiko Stuebner <heiko@sntech.de>,
 Peter Geis <pgwipeout@gmail.com>,
 Nicolas Frattaroli <frattaroli.nicolas@gmail.com>,
 Michael Riesch <michael.riesch@wolfvision.net>,
 Ezequiel Garcia <ezequiel@vanguardiasur.com.ar>,
 Alex Bee <knaerzche@gmail.com>, Liang Chen <cl@rock-chips.com>,
 Sascha Hauer <s.hauer@pengutronix.de>
Subject: [PATCH v6 3/5] arm64: dts: rockchip: add cooling map and trip points
 for gpu to rk356x
Date: Wed,  9 Feb 2022 22:55:47 +0100
Message-Id: <20220209215549.94524-4-michael.riesch@wolfvision.net>
X-Mailer: git-send-email 2.30.2
In-Reply-To: <20220209215549.94524-1-michael.riesch@wolfvision.net>
References: <20220209215549.94524-1-michael.riesch@wolfvision.net>
X-ClientProxiedBy: ZR0P278CA0197.CHEP278.PROD.OUTLOOK.COM
 (2603:10a6:910:44::8) To HE1PR0802MB2426.eurprd08.prod.outlook.com
 (2603:10a6:3:e1::23)
MIME-Version: 1.0
X-MS-PublicTrafficType: Email
X-MS-Office365-Filtering-Correlation-Id: 55b43b3c-cca9-4b47-99a0-08d9ec16f9f4
X-MS-TrafficTypeDiagnostic: VI1PR08MB4575:EE_
X-Microsoft-Antispam-PRVS: 
 <VI1PR08MB4575170E4C8D9A18DEBE9489F22E9@VI1PR08MB4575.eurprd08.prod.outlook.com>
X-MS-Oob-TLC-OOBClassifiers: OLM:4714;
X-MS-Exchange-SenderADCheck: 1
X-MS-Exchange-AntiSpam-Relay: 0
X-Microsoft-Antispam: BCL:0;
X-Microsoft-Antispam-Message-Info: 
 oFA9FRdjMaR5hNb3t8cmAfCr2rtDH6SWQhL1pX+VZZNUxDCiffvQv5LY7czI7PHZgPotHousvIA0qn/aWyz/KwY3m2jSZvmVwHAzsUEdVwrt7jo8Dlk1cVzhkR0xaIOIhP6cLJMjwdVhpBJXX08mAnGK03KCAXyX3hws+hxAkoHFB9ijFyC8fkAz0AEisLGRclDtIpFO6ovsbM7hjHUt9f4isop3CcHbf8sIxrSU/G/jZ14GR3jAMdXZz6OFBEiV5zhlA4Bmb62hxdM3azLOS8h6JmXRfEoniEHZsHsOS8P+LorOG3OlVf4Bw9YQP0L6bovPQPzbUM69731pxhFAsd8HpXLNJJ9ZZtBuReml2cKJoYF1bOKQtrQkl7YlxZuQS0fuX+73R4HvOHkR85bOT6c74rlIl1v4UR+nw4OeeveZ1EoKbze1T2yTt/abQdaBvTbei1f0qHlcdrTEL3ik6tK7+MFwvhg91MCRMIpXkpyVRQsg6BILX3fMjWGeFj6KIN/bqv8i729X7+k4rfGHHVN3gxES4hY28kZOOVuzAokrgvM6To8HK6XBHXB9LtBy34xI64JC5W4eZcDcNyu708kTlJAH8YM6IAxIxQ+gJNd8z32dzPisa/1t8XeTnzhHUrt4WyIwVN7WLMUeOsIZTg==
X-Forefront-Antispam-Report: CIP:255.255.255.255; CTRY:; LANG:en; SCL:1; SRV:;
 IPV:NLI; SFV:NSPM; H:HE1PR0802MB2426.eurprd08.prod.outlook.com; PTR:;
 CAT:NONE;
 SFS:(13230001)(4636009)(366004)(6512007)(38100700002)(2906002)(4326008)(508600001)(316002)(54906003)(52116002)(36756003)(6666004)(6506007)(6486002)(2616005)(44832011)(1076003)(86362001)(7416002)(66946007)(66556008)(186003)(8676002)(66476007)(8936002)(5660300002)(83380400001);
 DIR:OUT; SFP:1101;
X-MS-Exchange-AntiSpam-MessageData-ChunkCount: 1
X-MS-Exchange-AntiSpam-MessageData-0: 
 33HCziqWOjk0BomOiFUO1/1Yzn7mmsOC0LGjo0t6bjMrMuytrmfNUGJDdEDejyK9oZ2OBMe2b8UvKMptKTHyETU5x354mzGohOFTyRRb0/6TrVyFGZKlNo+R8v5DG1ajr3eHThKOX3JpsUESxV93TY/6ASVvhDMt3YGsAJNEC0a06mG/mo+9ECF7mvr+iE5fpXIuvNeXd7BUUW1ZHPb5lyxZQ4fqD43n0kbOnu2SVf+T+I7fJVHAI+z9WSbdOauZFBN2gNr/HoTHsT1QjZK1XwPKZgrP6B1OucjW868048hhlZmivUDu7fMzWGUtpWgQTtPzlF2bFNKV6FRzoP1MhLpgLn5/1vdhcGoH3nlGGWlbz4hrl1DpQ/5Ztzd/fX7rq7GY6fWc2KyLak7pLkkGX9J4qdC5CtS2nrwe2Rk5a4zZ+vgv+YZD7gfGCpdMRXeWrnP35kti02PFX3SeUz/ICiHr9hBYG56LiC92YflalDYKbqvHRLmaCgExtqFyBvY4nK+2FavmYuoARfV/EU+pexzU7HP/MKGxdyNpTfz3o2KUEawIf2s4imimMOO2uSzyXs1+OY+WoxZnA3GMDwnMmU3rkEe+g9s/I5Lcudgr26NuTHm8ikJTE+4kgSPfFjzDqHgu9r55YtQLFOwRogZjBlXEVtDJWCSo3om7yTpFZGBxQRX8ioVC4R1qYeGUbkY1BJSFZN/sMnnAvYQ7JmmnbGG1sy5A9MV9a5G6Y1/fbfzhclQH2zORdwuRPwyZajkYM4o3WXLe4+8wQt1oOvqxcibC7YvAeOQ2+1EloLyQ7ceJox7jxbk6yFyvNsQNwqplZD0w+dDjH4N2xGjR9pzPHlWongLUEvq2ZDllkQUo3oO7wOoy1KIyXn8TLc2YLSp/mu5usQzFN3nqarw/hfjOcjZ/vE2fkBPsVqymKhI//s2ELBvzyYCb9QPrc5EBLHZP/N9HW/mt0Zo6R0Bd2qJpYEqFEBPVEGrx9SYWOzqF7fMNTUUPA1i88qIrsgSrkoLcXBHtJZjrHLxo2UiaLqOZDXNmJGsIofwjExtQFinCbajIwzXLcu67Km+yqNBIM345jinSpbgcQWqeMzaMFxnObUQmTACCYqdCCwv6RsSqinw9VSfY2WCF5fEbh/P/C0MxHzbAaBGs8yNAnNb5X8Z/78SfKfeis1GNmIwmQCGL/ZYVjV0em9QZDqKjU7dNRWxTuQhWCbJxRS4oLGIyLEtWEg2UkztQg7smRQRoPOaP+5cdKoOlcGM5KkrfbmdpMdjjKzmbIR0rbBoMi0qQQSS4aGofplvMxTxzw6tQvHTPoilR2HiHHdSpbFapgirnE0ZTeeXpcEI7XyvjjUNB1fq/97PCconaHxicy8QFpb8O93AEa3TWBiQcf/NBCLZwFe4tVdoJ3iWRDLgSJymx7BdvdZFF3jxpFsHMqnDH2yeok+5x1OBtsceugulPXfmtxaEGoyffiFnzc+Yphn6W9EuZ0Va7B0z3GCMffjkVcve/kdxogfx7G8CPd9sh1tpZTXtK6j8FnUw7cP56gSY1QYDD/3jxzplCCj3+hfElonDT00rpu0XrW6BJbAmvT5n3J8xqUiDP4M/6DCRJbBPHfpf7mtCZGpQ24gPY2weR/e00X26aK5LRU1KXZ8HdBrhGfrRtBNJHQbkF3Iezjx8T6CDGg6UMXfkuHn0wygE5aIPr25ug3aXl/Crq/cMeVts/P4JomMKB0VfKoSX6bnCJk4oUAQQ0nAUGYEtJi9JfS5AQNMg=
X-OriginatorOrg: wolfvision.net
X-MS-Exchange-CrossTenant-Network-Message-Id: 
 55b43b3c-cca9-4b47-99a0-08d9ec16f9f4
X-MS-Exchange-CrossTenant-AuthSource: 
 HE1PR0802MB2426.eurprd08.prod.outlook.com
X-MS-Exchange-CrossTenant-AuthAs: Internal
X-MS-Exchange-CrossTenant-OriginalArrivalTime: 09 Feb 2022 21:56:07.8815 (UTC)
X-MS-Exchange-CrossTenant-FromEntityHeader: Hosted
X-MS-Exchange-CrossTenant-Id: e94ec9da-9183-471e-83b3-51baa8eb804f
X-MS-Exchange-CrossTenant-MailboxType: HOSTED
X-MS-Exchange-CrossTenant-UserPrincipalName: 
 SrWl0wR9OWIqqS9VlNzgqjLU/302gs++zhv+WGpdqejbcx4rmucXfCF1+CITS1AX553xuN2dXgq2NXbbn8IOch27Uxzxls7MftpsIi1anow=
X-MS-Exchange-Transport-CrossTenantHeadersStamped: VI1PR08MB4575
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20220209_135613_741870_FE1B44BB 
X-CRM114-Status: GOOD (  12.36  )
X-BeenThere: linux-rockchip@lists.infradead.org
X-Mailman-Version: 2.1.34
Precedence: list
List-Id: Upstream kernel work for Rockchip platforms
 <linux-rockchip.lists.infradead.org>
List-Unsubscribe: <http://lists.infradead.org/mailman/options/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-rockchip/>
List-Post: <mailto:linux-rockchip@lists.infradead.org>
List-Help: <mailto:linux-rockchip-request@lists.infradead.org?subject=help>
List-Subscribe: <http://lists.infradead.org/mailman/listinfo/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=subscribe>
Sender: "Linux-rockchip" <linux-rockchip-bounces@lists.infradead.org>
Errors-To: 
 linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org

From: Alex Bee <knaerzche@gmail.com>

RK356x SoCs have a second thermal sensor for the GPU. This adds the
cooling map and trip points for it to make use of its contribution as
a cooling device.

Signed-off-by: Alex Bee <knaerzche@gmail.com>
Signed-off-by: Michael Riesch <michael.riesch@wolfvision.net>
---
 arch/arm64/boot/dts/rockchip/rk356x.dtsi | 27 ++++++++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk356x.dtsi b/arch/arm64/boot/dts/rockchip/rk356x.dtsi
index 50bbea862a6a..37194d735028 100644
--- a/arch/arm64/boot/dts/rockchip/rk356x.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk356x.dtsi
@@ -1093,6 +1093,33 @@ gpu_thermal: gpu-thermal {
 			polling-delay = <1000>; /* milliseconds */
 
 			thermal-sensors = <&tsadc 1>;
+
+			trips {
+				gpu_threshold: gpu-threshold {
+					temperature = <70000>;
+					hysteresis = <2000>;
+					type = "passive";
+				};
+				gpu_target: gpu-target {
+					temperature = <75000>;
+					hysteresis = <2000>;
+					type = "passive";
+				};
+				gpu_crit: gpu-crit {
+					temperature = <95000>;
+					hysteresis = <2000>;
+					type = "critical";
+				};
+			};
+
+			cooling-maps {
+				map0 {
+					trip = <&gpu_target>;
+					cooling-device =
+						<&gpu THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
+				};
+			};
+
 		};
 	};
 

From patchwork Wed Feb  9 21:55:48 2022
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Michael Riesch <michael.riesch@wolfvision.net>
X-Patchwork-Id: 12740948
Return-Path: 
 <linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by smtp.lore.kernel.org (Postfix) with ESMTPS id 92174C433FE
	for <linux-rockchip@archiver.kernel.org>;
 Wed,  9 Feb 2022 21:58:11 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20210309; h=Sender:
	Content-Transfer-Encoding:Content-Type:List-Subscribe:List-Help:List-Post:
	List-Archive:List-Unsubscribe:List-Id:MIME-Version:References:In-Reply-To:
	Message-Id:Date:Subject:Cc:To:From:Reply-To:Content-ID:Content-Description:
	Resent-Date:Resent-From:Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:
	List-Owner; bh=72BTnSlFHZx0QeOW+dmt4coKkMsNPhiIW4iD0ynnxaw=; b=qZ44CSRPc6vCHY
	pcBw6ZX7UiX/mGIxH5C298RmAJmG83LhShR63aJ0CjX2pRaZC0QKsRBhE1aaIlsHtQrc+mOiu6qZN
	gujQiVzA6tD3cPGMohuAbRgnhWNI3fj6OHAY78bXkkLbGcd9bcB0rPuDWNY0eV8Y1mYuTjurfrmRs
	ITFxHuL0hhiTsAdpOKGFDygzTRjzy85+EnO0ZSHxgIO5J2aAKQcevNZHzSlvMG6EYpwAQLyqdznt3
	84yDp89BEebuA+BH9WBhXnRM1NHpPRewm4tDHWGD2bpLz5OMgV24Z7N0BHxyfzy9AY0S/ILeOKhcr
	4xoPkLCn92oyoZPSJLdg==;
Received: from localhost ([::1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.94.2 #2 (Red Hat Linux))
	id 1nHuyW-001lON-2G; Wed, 09 Feb 2022 21:58:08 +0000
Received: from mail-ve1eur02on0600.outbound.protection.outlook.com
 ([2a01:111:f400:fe06::600]
 helo=EUR02-VE1-obe.outbound.protection.outlook.com)
 by bombadil.infradead.org with esmtps (Exim 4.94.2 #2 (Red Hat Linux))
 id 1nHuwh-001kMp-Gl; Wed, 09 Feb 2022 21:56:17 +0000
ARC-Seal: i=1; a=rsa-sha256; s=arcselector9901; d=microsoft.com; cv=none;
 b=V5yzLHvTyFV5X3lRwSM50NV92otEgcOM+o21oZHvveMcKRgAlLGN7DMz3BguHCCSLOOGabgBH49P/r+JitpCRaxILbM+Juzi5mjInu0i3mm0dYZDznEVkq6/+bWXjoBCF3c9B8lwcOQ8Ygo9VPmWXkjkpyiizdAEjj6LIcI/qeLLb3Ci+DoXwHE00ROVMqEAgNJzDiM6h56jrQYHC/rmDH1Zj0++BiygBSJ9oGeo8M0r/D16jfQLD3Encle4fz6WSygoUerfTIMq7TrrfdJ5U9m+aZkIKEN0+oGMjEOQ/vL055g0UZMyDOKfumLqrBQehKwX5ahS5NjaBv+eaUQVyg==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector9901;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-AntiSpam-MessageData-ChunkCount:X-MS-Exchange-AntiSpam-MessageData-0:X-MS-Exchange-AntiSpam-MessageData-1;
 bh=nXhbRDQGkISr6kHJqbllfTp+stN+je4oakh0iU0AGdo=;
 b=bZa6kUGDNC1dFRKOYnugbEDK5BoowARKNzbtxxie6OWdAUpvBMUNrV4onqiexJaye3KIRJWjLa1fEub3eL70q8vv3IyLHf+d5yzlQNTbXompoxTBD4Dv8RFifth5crm7ihpjauuoyt7Ixrk4pubUrwJEtTPeypdFhZQysZRvPqGSBjl7EcetiQCICXPOwglYlHM4A9ql8Eb/Y9AJx26yno6rMIwh2KyaIILehcdVqF2oxMsK8Gfeqxxe8/lRrI3yJs76LWhrN7qZSVceEuxu8XN4g+WmVxwksyeyVwOYFLUKQ4Gw5QUCw0BDooXNG6QP+ssf3IrHfr1Mzet3zDUR5Q==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=none; dmarc=none;
 dkim=none; arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=wolfvision.net;
 s=selector2;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=nXhbRDQGkISr6kHJqbllfTp+stN+je4oakh0iU0AGdo=;
 b=tVoFmdo1+vr99MdIsV/qMaCMYPP/hwKsHOdffuV7cgN86qa3wiGNK4S9Wc2r6Q3IDC6frcZ4mcZugd3u3rXBZIKgdx7NqDt+C4/3FUkKAZY5jaGvxmt0Le8qOFeuh5JGITur+1hb11X8TIikLvhRUdz889IJFnr2eFeGRF7gIFg=
Authentication-Results: dkim=none (message not signed)
 header.d=none;dmarc=none action=none header.from=wolfvision.net;
Received: from HE1PR0802MB2426.eurprd08.prod.outlook.com (2603:10a6:3:e1::23)
 by VI1PR08MB4575.eurprd08.prod.outlook.com (2603:10a6:803:eb::26)
 with Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.4951.17; Wed, 9 Feb
 2022 21:56:10 +0000
Received: from HE1PR0802MB2426.eurprd08.prod.outlook.com
 ([fe80::e14f:197:5001:4c56]) by HE1PR0802MB2426.eurprd08.prod.outlook.com
 ([fe80::e14f:197:5001:4c56%3]) with mapi id 15.20.4951.019; Wed, 9 Feb 2022
 21:56:10 +0000
From: Michael Riesch <michael.riesch@wolfvision.net>
To: dri-devel@lists.freedesktop.org, devicetree@vger.kernel.org,
 linux-kernel@vger.kernel.org, linux-arm-kernel@lists.infradead.org,
 linux-rockchip@lists.infradead.org
Cc: David Airlie <airlied@linux.ie>, Daniel Vetter <daniel@ffwll.ch>,
 Rob Herring <robh+dt@kernel.org>, Heiko Stuebner <heiko@sntech.de>,
 Peter Geis <pgwipeout@gmail.com>,
 Nicolas Frattaroli <frattaroli.nicolas@gmail.com>,
 Michael Riesch <michael.riesch@wolfvision.net>,
 Ezequiel Garcia <ezequiel@vanguardiasur.com.ar>,
 Alex Bee <knaerzche@gmail.com>, Liang Chen <cl@rock-chips.com>,
 Sascha Hauer <s.hauer@pengutronix.de>,
 Ezequiel Garcia <ezequiel@collabora.com>
Subject: [PATCH v6 4/5] arm64: dts: rockchip: enable the gpu on quartz64-a
Date: Wed,  9 Feb 2022 22:55:48 +0100
Message-Id: <20220209215549.94524-5-michael.riesch@wolfvision.net>
X-Mailer: git-send-email 2.30.2
In-Reply-To: <20220209215549.94524-1-michael.riesch@wolfvision.net>
References: <20220209215549.94524-1-michael.riesch@wolfvision.net>
X-ClientProxiedBy: ZR0P278CA0197.CHEP278.PROD.OUTLOOK.COM
 (2603:10a6:910:44::8) To HE1PR0802MB2426.eurprd08.prod.outlook.com
 (2603:10a6:3:e1::23)
MIME-Version: 1.0
X-MS-PublicTrafficType: Email
X-MS-Office365-Filtering-Correlation-Id: 84f16967-ab7a-44b3-22e8-08d9ec16fb28
X-MS-TrafficTypeDiagnostic: VI1PR08MB4575:EE_
X-Microsoft-Antispam-PRVS: 
 <VI1PR08MB45759822CE8B0FF3AFA967FDF22E9@VI1PR08MB4575.eurprd08.prod.outlook.com>
X-MS-Oob-TLC-OOBClassifiers: OLM:983;
X-MS-Exchange-SenderADCheck: 1
X-MS-Exchange-AntiSpam-Relay: 0
X-Microsoft-Antispam: BCL:0;
X-Microsoft-Antispam-Message-Info: 
 HEboKU0RxdD3//Zwd58TyMJXtXfLaifgiJ6EPe6YsN1/tMqNgnx3Nc9WhH+6gut5qCx01scuEQtYAZ8O0GUN1Sj1QHzE0OwCWmhgtowHM/nAnAxkzmREiwbdnIPehlM5JPBotWtNKC79vZUezvYGjowWVXetNxS4Kx8SPWtwlH6ZHG26Ng06zkBSqAUCs0k2GtkRFi8I6o6hpEvl9qPGLfauvLYRB9So8112J4GEMCyG7i803j24PnZ0YdOps4djf5KC0pj2pOrtwD3+G19ooyIm6SjZi+B5Mu5V7p2ko6k4f+cOUyrJ8y3JeO2VVtRm7MfwmjE1QZ+90EQXeSHtjl/V0nfFl/B3ScWEqMXRQH5Mq1Jo90xHspYQd0+1hMCP3LXHIVDOpSbIPkUFCDYhgdrSmg2EjNQwblCCa5/bdjlI9VCL3EV3jzxyhg04mLkHKyBK/MGTdw+ED+zkIMtQV77vOq/XnioVs9lcJdsmyCeIHBmDjVo6drQIun5kJxblvBPWiPZQqt2rckPsLEu5hu+kkbOkgOdjzoYWRCj0wqF8uEs3ybDujDsX59RyfX3j++CACJD9V2oyCMiTkfHd9YLzT36CjFQZDorssrbuMtGQPGrkJ+3mujNgsKbyvBUz/fpNF4SL1bl4WGOQ50kqyQ==
X-Forefront-Antispam-Report: CIP:255.255.255.255; CTRY:; LANG:en; SCL:1; SRV:;
 IPV:NLI; SFV:NSPM; H:HE1PR0802MB2426.eurprd08.prod.outlook.com; PTR:;
 CAT:NONE;
 SFS:(13230001)(4636009)(366004)(6512007)(38100700002)(2906002)(4326008)(508600001)(316002)(54906003)(52116002)(36756003)(6666004)(6506007)(6486002)(2616005)(44832011)(1076003)(86362001)(7416002)(66946007)(66556008)(186003)(4744005)(8676002)(66476007)(8936002)(5660300002);
 DIR:OUT; SFP:1101;
X-MS-Exchange-AntiSpam-MessageData-ChunkCount: 1
X-MS-Exchange-AntiSpam-MessageData-0: 
 /ldw58wNDPW4tNZuxLTB9D6PdJgOT9t+3sE2qTOlSAAdE6Xi8KAiSAbc+xaWNyEtK0EshxqI4zMGq28O2PO+9Y77kg4qKETcnylwx2GKAq/z9m1JoVTTvi2gpmUqC3Rw7DpFu0EmTgom2Bqs7CICta9MheJlVNu90mOK2rlM1TmNa9qd2KcbjKTIkbPJg824g/dFT+GNbdVKX4kLTzTSN/UDmYMhZbpTzCsZT7Pi7B8qPazWBR6dMmd2gqg0Io6HRNlYJnmcExEgKQyR7SFwcnIEbLC85/8QprYZR77xnLWaAkqJo/YMzEgCURIm1B9v9KBkh414WChEjZ7BC8mk/KYNy7T6ci4hIqwx8us4qoMW33D3ZYL/G1yy6qRgwbtlgeHZHtr1yOqcpD0r3vgnAudMiW7zktcmXx/zJ9R/tqzncYHd+R6tIh3LsxgWeELRBe0Zc0hDeo6ajstMzvmfMyaqaH3gV5ZBCVPyhrPKywC/zM4N0RjVNBWkxpH9v070EFWfxUU+9PPaPTY0YH424hJduDcjFycxyX38SsePVYpGT0uqyvGtaWf+gvf0hjt9cLwTy2urcoh/t3mBoMhPY2adr86B9DRAWdqHd6xKXH/uLXu0w1Y2ULaw8GhdNGHA1mff1gnSsZplwL2cu5VD/N85J9qXXECEkdLASnJ5LzuC6tOEK/xlNSzFgy6omtczKFDKs/M5aqCU+QvOWiPX+voPQ0WYRp7ilL+LxScAqLil5Siy/nLCBfEkbjo2Q1YRGajCmqDWURUX09rZDhCwIHJ8/mJdfUnpcmoPPGMM4MEXVIjl1JKUoToO3Q5Q+jz2ARDRjjlyz5mTIDlFiAiRFk/xjxGrEQRX6IQMgkIg3WRqREe9KVbBQsWpDeEBnBzl6XWQM+VgnYIEBCGAsLaCAmSUhp0ycdYRApaNtXd1t9PauM6kp7tNf/OHgIvZEQvNbh3wjqZLr2aWqqCmt1j3YRuteTV+g3/QOLSSU8I4OYbFyo1XWl+knK2beuH4d1adg7VlocLMv9WWS6GLqHyWs/vBmEFtVVX45uk6DgjTv8oepxxhkl5G+16pi+/CKI1TudxXIp2PZHmJ0titQ9r4muh4vHsB/cEml1YogGvM9VPkHr7Xi3bOzYO+aDnPPdNgFyaNww6d+Nb2iv+Byl0MPOQRpnrmZM1Kba0kEeQmTp7JPvdZi1ljdYAb1SqBHfPmGOSHLs1hA6SzdSJuwmJ+o+XYgpm+qszqx68rRdm0nu8OJETcaGgY5CjLa6e0T3rfrrAyZfMlS2PmmSpS14wCgKx+CM+GUZEo4AqsepCzcTZTWEthsl0klWERvaI0ddl2nooDEfZ9+rxZIiJ6x2mBsmQchQFB31meM4wpMpHlZzcFnIZP7RjNz0sDckhbq5dmPMskMroImVAXbrGcED57KTYe5bF7mgu74PkRFlT4S1QZoSCa3nJ5jgu+W8LhlbItIM93pMAVdZbPU66LyGzMXmlDR058U4VYHRdiBHrgHWEUKDQn7fAxZBaC5tQl4+VnvOAHwwi75wWjwQyFP23EAY5zYhud8VYo1zxlyhbBkvbvi3BNzML8L9QvDcJH1s3M+EBbSsLvK5TQYlWoj6Xsl0vZTENzJEssKOsAsj20ExvRrdpybtB58BbIte4VtkWM24WZmQo1sdhPm9uWXw5aNtYPSXRmoFiNIHPL0qQn3GQIH6CBCU3K2BczqyMVN1k9pjc+3NO6dXnYgM6QAWDGBV2cGeJDjxcu9RBeki9VgO0=
X-OriginatorOrg: wolfvision.net
X-MS-Exchange-CrossTenant-Network-Message-Id: 
 84f16967-ab7a-44b3-22e8-08d9ec16fb28
X-MS-Exchange-CrossTenant-AuthSource: 
 HE1PR0802MB2426.eurprd08.prod.outlook.com
X-MS-Exchange-CrossTenant-AuthAs: Internal
X-MS-Exchange-CrossTenant-OriginalArrivalTime: 09 Feb 2022 21:56:09.8657 (UTC)
X-MS-Exchange-CrossTenant-FromEntityHeader: Hosted
X-MS-Exchange-CrossTenant-Id: e94ec9da-9183-471e-83b3-51baa8eb804f
X-MS-Exchange-CrossTenant-MailboxType: HOSTED
X-MS-Exchange-CrossTenant-UserPrincipalName: 
 d3ENJpxdhu817cz/B5eN+sGxv8i7qdmldN75GIjt12Lt3gNbl9vPVGqH/0pVxoxsmDTyeaNy53FCpQQlicxBKkmD1udyphBmu0KNGSb86jg=
X-MS-Exchange-Transport-CrossTenantHeadersStamped: VI1PR08MB4575
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20220209_135615_624767_C9049C5B 
X-CRM114-Status: GOOD (  10.02  )
X-BeenThere: linux-rockchip@lists.infradead.org
X-Mailman-Version: 2.1.34
Precedence: list
List-Id: Upstream kernel work for Rockchip platforms
 <linux-rockchip.lists.infradead.org>
List-Unsubscribe: <http://lists.infradead.org/mailman/options/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-rockchip/>
List-Post: <mailto:linux-rockchip@lists.infradead.org>
List-Help: <mailto:linux-rockchip-request@lists.infradead.org?subject=help>
List-Subscribe: <http://lists.infradead.org/mailman/listinfo/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=subscribe>
Sender: "Linux-rockchip" <linux-rockchip-bounces@lists.infradead.org>
Errors-To: 
 linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org

From: Ezequiel Garcia <ezequiel@collabora.com>

Enable the GPU core on the Pine64 Quartz64 Model A.

Signed-off-by: Ezequiel Garcia <ezequiel@collabora.com>
Signed-off-by: Alex Bee <knaerzche@gmail.com>
Signed-off-by: Michael Riesch <michael.riesch@wolfvision.net>
---
 arch/arm64/boot/dts/rockchip/rk3566-quartz64-a.dts | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a.dts b/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a.dts
index 3e65465ac7d5..b048db6cff3a 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a.dts
@@ -221,6 +221,11 @@ &gmac1m0_clkinout
 	status = "okay";
 };
 
+&gpu {
+	mali-supply = <&vdd_gpu>;
+	status = "okay";
+};
+
 &i2c0 {
 	status = "okay";
 

From patchwork Wed Feb  9 21:55:49 2022
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Michael Riesch <michael.riesch@wolfvision.net>
X-Patchwork-Id: 12740949
Return-Path: 
 <linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by smtp.lore.kernel.org (Postfix) with ESMTPS id 80D52C433EF
	for <linux-rockchip@archiver.kernel.org>;
 Wed,  9 Feb 2022 21:58:51 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20210309; h=Sender:
	Content-Transfer-Encoding:Content-Type:List-Subscribe:List-Help:List-Post:
	List-Archive:List-Unsubscribe:List-Id:MIME-Version:References:In-Reply-To:
	Message-Id:Date:Subject:Cc:To:From:Reply-To:Content-ID:Content-Description:
	Resent-Date:Resent-From:Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:
	List-Owner; bh=cRkgc0NChP4C4BiqVh/HyIGJ2B/Y7xbtwiVoebR55Jw=; b=CI+mzgF5/eG53B
	Mmj1coNkV62oS7MEmlL/agvnWVJjWQiH0oDHYxEOLJvmZimbPFTGT7o+Dli5NRqv3GA8lnNCZY3Q3
	sPbRxpyvcrEiK+ymTSNxNphU4D09xsllvM1ZEUd/fN57rsDZTs87nKw4eKaWC6zvi/hD2h/X3ohjT
	8zSHw7T2EO90De8Irkx+pdy+ZBfSZ+7v8As9SLH5Trf8lciKD+dVZu2SNRo0rhvgsOpk7vBGNtxXo
	wja/lIAFYC4WQHGlnuGrh+TpZK48WUNOwY1MmWi1QilsEXkukPvCLmz4P+TUyIaPZOy+PLqUS0Bbs
	VjZJucCLLHfnsR6MfFMw==;
Received: from localhost ([::1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.94.2 #2 (Red Hat Linux))
	id 1nHuz8-001lfX-VQ; Wed, 09 Feb 2022 21:58:47 +0000
Received: from mail-ve1eur02on0600.outbound.protection.outlook.com
 ([2a01:111:f400:fe06::600]
 helo=EUR02-VE1-obe.outbound.protection.outlook.com)
 by bombadil.infradead.org with esmtps (Exim 4.94.2 #2 (Red Hat Linux))
 id 1nHuwj-001kMp-F9; Wed, 09 Feb 2022 21:56:19 +0000
ARC-Seal: i=1; a=rsa-sha256; s=arcselector9901; d=microsoft.com; cv=none;
 b=mel02+VPhLCRpZVKfJ3uH/zinH8fI3ruY7AszZjam/xqGs79OQrw2MOmLNqlsP8NjyGT7YQsozPwTkDaBkwQrws7qM9uMR2GwjZgDUpn9Cp1C4FaZLkMn9P/twNIjhyYL6ENHVy9JuQO4rMruemBvyl/3YzwC1D+9NhL67e/bIPaXQENy3reOO4ZEdYp7qIbUbZGjvTd0H5qIMtn22RDpru116E9AbSWvCKuvgItvd5yawBY9BA6kHQJZmJayBzA9sAcyS8KGnJy+vzOiovHteLDEA73gkz7jm+05my/30RKF9j2Qld3fKMW0vBL4DcN8119oF3meMw1EUIiZbdwCQ==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector9901;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-AntiSpam-MessageData-ChunkCount:X-MS-Exchange-AntiSpam-MessageData-0:X-MS-Exchange-AntiSpam-MessageData-1;
 bh=maYIBthui/wiXVu7QZ+ik0rjmMzwVfPApPwcKGj9fFM=;
 b=Kk/BTX6FV1ENIOJpsiM7kWtpubDE3jCG4TxJMnto6gpooqIz9fqfD6OsD2kIGhe0n5eG/EEUahKvgiZEXvk0N1Fm6/SoTXeTogPg6v9m9scBpBIqKz7LkEnFo6KXmqSUeQkbjk6nQqiyrVu1xgvB3VSdAMqjpLY330wZJiZ7o1DWPdlpnQ+PVW0n7qOH/TTM19DbGEtmK6SPXZjsAFqu/HFXS9RPFK4843RyfeX46EFdfYT5c1uEY7/7JwixQJEbbiX1NmZFX4jWPKDPJMrQ7dvznZ0vUPj3u3J3F+hCl5J4NVR5pUS+o8FwEpRLZMUkwoAjbWMXNBk888J51F5Ymw==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=none; dmarc=none;
 dkim=none; arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=wolfvision.net;
 s=selector2;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=maYIBthui/wiXVu7QZ+ik0rjmMzwVfPApPwcKGj9fFM=;
 b=lnoDTr5+OQQRPID+rxkYWRGDwLoxoXeM+2PZ8pSHt0OFrlnDgTvfX8dNBJdDPy1sG9S7zlB5qqXYrUWKeaPXDNaJFpWNMiUCW9kzf07m3KrPU+DiLQXJjmOCSoOQmxrGXEm9A7ccD2ZYMOzggN9kEsnuO6i5RutIDezV3AzXElI=
Authentication-Results: dkim=none (message not signed)
 header.d=none;dmarc=none action=none header.from=wolfvision.net;
Received: from HE1PR0802MB2426.eurprd08.prod.outlook.com (2603:10a6:3:e1::23)
 by VI1PR08MB4575.eurprd08.prod.outlook.com (2603:10a6:803:eb::26)
 with Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.4951.17; Wed, 9 Feb
 2022 21:56:11 +0000
Received: from HE1PR0802MB2426.eurprd08.prod.outlook.com
 ([fe80::e14f:197:5001:4c56]) by HE1PR0802MB2426.eurprd08.prod.outlook.com
 ([fe80::e14f:197:5001:4c56%3]) with mapi id 15.20.4951.019; Wed, 9 Feb 2022
 21:56:11 +0000
From: Michael Riesch <michael.riesch@wolfvision.net>
To: dri-devel@lists.freedesktop.org, devicetree@vger.kernel.org,
 linux-kernel@vger.kernel.org, linux-arm-kernel@lists.infradead.org,
 linux-rockchip@lists.infradead.org
Cc: David Airlie <airlied@linux.ie>, Daniel Vetter <daniel@ffwll.ch>,
 Rob Herring <robh+dt@kernel.org>, Heiko Stuebner <heiko@sntech.de>,
 Peter Geis <pgwipeout@gmail.com>,
 Nicolas Frattaroli <frattaroli.nicolas@gmail.com>,
 Michael Riesch <michael.riesch@wolfvision.net>,
 Ezequiel Garcia <ezequiel@vanguardiasur.com.ar>,
 Alex Bee <knaerzche@gmail.com>, Liang Chen <cl@rock-chips.com>,
 Sascha Hauer <s.hauer@pengutronix.de>
Subject: [PATCH v6 5/5] arm64: dts: rockchip: enable the gpu on
 rk3568-evb1-v10
Date: Wed,  9 Feb 2022 22:55:49 +0100
Message-Id: <20220209215549.94524-6-michael.riesch@wolfvision.net>
X-Mailer: git-send-email 2.30.2
In-Reply-To: <20220209215549.94524-1-michael.riesch@wolfvision.net>
References: <20220209215549.94524-1-michael.riesch@wolfvision.net>
X-ClientProxiedBy: ZR0P278CA0197.CHEP278.PROD.OUTLOOK.COM
 (2603:10a6:910:44::8) To HE1PR0802MB2426.eurprd08.prod.outlook.com
 (2603:10a6:3:e1::23)
MIME-Version: 1.0
X-MS-PublicTrafficType: Email
X-MS-Office365-Filtering-Correlation-Id: 5732fd5d-8de5-46f5-8747-08d9ec16fc35
X-MS-TrafficTypeDiagnostic: VI1PR08MB4575:EE_
X-Microsoft-Antispam-PRVS: 
 <VI1PR08MB457563BC741397E91CA2D8A5F22E9@VI1PR08MB4575.eurprd08.prod.outlook.com>
X-MS-Oob-TLC-OOBClassifiers: OLM:983;
X-MS-Exchange-SenderADCheck: 1
X-MS-Exchange-AntiSpam-Relay: 0
X-Microsoft-Antispam: BCL:0;
X-Microsoft-Antispam-Message-Info: 
 aNjspTOzH+dT/7ElZ/ySJ/XKm+MYxBMw1KL3IF4OMkytRhURNqz8gdWn41V2qKMz2G9KdIrMHEqq5dzksOEkYo+F6/TIRb6oY3kNSIcBNRbLv11NFitiqanaxFo30UEXL63aSe1zBM6R1LkFmcv3r5otDFNLKkF3vqnHT06apLS3AFqXtVFbUKEbVZtplAboafQh7IMfl1CI+GxGLzegnIDF8ld68iqB94tA8K3VBnu0hT3zZGCXAtSrEbhUPurfYsWhaHXiWfRtPfZ79Ouk/FaNIUWzT83Hs9v6dUFsiRc2FlbBw3esqzkiTQfbAgUU44gTKWPr2/AhslsbtB4w4j50JdqNYl80RbUqjb1zfQ4154Kjcx632ii2N+p6bjdBg0lyKxtW9JyWnEkLbCZH9sS++jwfruLlceRQpMzztW+XpvDXNW41ARDbEfZfs+lebBGBccbpsjmHxzTtpjeiw+5Zsw9Lybq/2/ShAaCvSb/MgliXIpoHqhfjgwJl/VJW+VHRRxASXN/Pm6c7mGD2cJqMrRL6dHZlg5LM8bv4Y2ps3dXAuHN0QvVbYN8lVlt0GQcT2mm0tTvqPR/6LKqs0Jaa8XLKwaeMF9GSEplrihYYR9SWSgjE+92ZZRl08jlo2+7ZwTTMi7Lh/JpefDQyKw==
X-Forefront-Antispam-Report: CIP:255.255.255.255; CTRY:; LANG:en; SCL:1; SRV:;
 IPV:NLI; SFV:NSPM; H:HE1PR0802MB2426.eurprd08.prod.outlook.com; PTR:;
 CAT:NONE;
 SFS:(13230001)(4636009)(366004)(6512007)(38100700002)(2906002)(4326008)(508600001)(316002)(54906003)(52116002)(36756003)(6666004)(6506007)(6486002)(2616005)(44832011)(1076003)(86362001)(7416002)(66946007)(66556008)(186003)(4744005)(8676002)(66476007)(8936002)(5660300002);
 DIR:OUT; SFP:1101;
X-MS-Exchange-AntiSpam-MessageData-ChunkCount: 1
X-MS-Exchange-AntiSpam-MessageData-0: 
 T1YckavoAlKkyxtImsZTJkAPNCF8FR8gflCwPSAGftNHedZ3qliLE2d7vrsTzQjECaJifGesmSElpc2Y1J1Puu32g/3CdZSjsHeyG8VZYeUAR2F/PURjkkZUAjckSPu4eKq8U2QG+Ouk1YU3x1tjkkFrMYA03If2Qv/dDT5rSbhZSY7j2qCWiQxZq50XEI1JQ3LOj/ujWZe5bvnGhVuRWuZKMh2y2423mthpobsNOpiqJiLWNJRiARYcbcSEXPX4Gmep9eWwzQXbhWNUL2nhzvuBuHiPsLVFhnKEceSI9wh6JFtrEZTZLSXVEzH19ZMLuSu6ce6j5fDUrKxTUQ67eyPSpnZw8oXkXss6u8AMgiN+Izk9llKcJgYptXRsdWm5GkSXEdmlhIc8kqmQStffVfMuicGQukJz7b+gwDQ3sZ5fN4FJ7TIgrF9Qy/Xe0qUnJbCFVFxdBQZxNNYR0E1XFzexEqMXFRg4WDfspxtUr4is6H77pJ5DsgBmrN3GKSnUiO9mz2TSgbf0HH9K1RDq+Hi3TXNZfsx5yGlyivL1Fh6gzIpZaUDCU6G6Nt2UZs72OXesiDOkHNSi8iQRqZtQomprisidDUqDaQl1sLVWpTKglQ3G7GFcCV0DbPzEsk2gmE5WUdnd82vILWe55rrlk2N4FulNwLR/PHzL8DfjfbddoleMLNlA8rmWLvDu9Y8PkdjQtU5Ml0E37BpB8AvMAOqzdLofmB0F+pMzHxS/EFAJ4qptdZ+brOUHSQoolUwq5T5JSjcfu4piIL7tJc6BldvAjC9VH28O+3u04mYqsGsiGpCtuoB6bEdpF9TZ3TXKGJHFn+2v4M3sYw3jiP2I2yGv0zolAn2iKSa3C/PHmwnPc48ZZ28mSZtXQOn4dIfhNCOTnhutAdD+9WyCmi5h9WMuaivtVGtmqDW08JTLtFCV7Ekaqlu4pcT8fs/J05m3AHIjtdyWGrcKeK9Lq2rfb4TqHn5sLM8lAYF3lN+5yk8/mOfEqoC4ZTz+l9O5p+uZzR4Xuk48oa2lJIiK45GUwtRmkCYpnJ4OWDURQHexumLiQvnXKRB9BkNgUV18/d6H7HyQvk7ipb+GUwlyaxEnYKkcd0mz3XL5ShyQA8IOHjx2ei9aUMKyq51FKOWDbtaom++yYdH+tKBjq5NhjcVDBk7VMnzNtzc9uTp4Uk531xY/T9OVgLOTZ207KSUIeoJXXzU8E3BzpPedislJMXpsGW1TRmjP5+aFlEMw5lKDpdpNycPlntp4pt3EjhWrFSnx8KRaE3idryiAX6KnuU8Qh52WTz8bwYVnaW5fBqyDvKd6sRl9+5zWMkFKynTxuYYraeLtOq/QI6sAFNaidDI90UiPnq0JQ7EI6OtzhobRObTaiAxViQ2xyCkZ7CqUll24oRTm+oT1pZpW5JubCNcKQhMTgyVaq3JsjWnQm+IeYWedsxe2fLvL0WqHJfmGsl2bsSNNGDUDyF1ACfgDRIa4EYydszo1ylY8O+3x7eartncmeOmw4FSCwhWhxQKBIlu9CiJnDzcY1czkYrGSc79XIxOLtvM8uX4W63mv9pahjk3xYEmCzPV+NjqmqwVknuTnCrhRHtaM6xE+5QjfdsvHC/RKtC6vq4RTfIX24AS9wNsBcA2pd6mQERED200yoEyiAUJrKroHjePP7E0AsY4OyqMWGrM3ms8nspuJIq+LVOQAX0ssWf1htb3S/bmM7lxc6qggcXws76ZSwWY/9C8ahkw1N+DPFSFXQll9SE/SNmg=
X-OriginatorOrg: wolfvision.net
X-MS-Exchange-CrossTenant-Network-Message-Id: 
 5732fd5d-8de5-46f5-8747-08d9ec16fc35
X-MS-Exchange-CrossTenant-AuthSource: 
 HE1PR0802MB2426.eurprd08.prod.outlook.com
X-MS-Exchange-CrossTenant-AuthAs: Internal
X-MS-Exchange-CrossTenant-OriginalArrivalTime: 09 Feb 2022 21:56:11.6469 (UTC)
X-MS-Exchange-CrossTenant-FromEntityHeader: Hosted
X-MS-Exchange-CrossTenant-Id: e94ec9da-9183-471e-83b3-51baa8eb804f
X-MS-Exchange-CrossTenant-MailboxType: HOSTED
X-MS-Exchange-CrossTenant-UserPrincipalName: 
 DhAwlU5YOwAfrLHtYv27SsV8NfCbLMPMuc+zR7rtt3hChzzXtNipIOCyaS+VItsubYr3IaEXdtG61pjQi70ms9jflbKmCe58evvDuHJ4cNU=
X-MS-Exchange-Transport-CrossTenantHeadersStamped: VI1PR08MB4575
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20220209_135617_586783_C1E62D09 
X-CRM114-Status: GOOD (  10.55  )
X-BeenThere: linux-rockchip@lists.infradead.org
X-Mailman-Version: 2.1.34
Precedence: list
List-Id: Upstream kernel work for Rockchip platforms
 <linux-rockchip.lists.infradead.org>
List-Unsubscribe: <http://lists.infradead.org/mailman/options/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-rockchip/>
List-Post: <mailto:linux-rockchip@lists.infradead.org>
List-Help: <mailto:linux-rockchip-request@lists.infradead.org?subject=help>
List-Subscribe: <http://lists.infradead.org/mailman/listinfo/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=subscribe>
Sender: "Linux-rockchip" <linux-rockchip-bounces@lists.infradead.org>
Errors-To: 
 linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org

Enable the GPU core on the Rockchip RK3568 EVB1.

Signed-off-by: Michael Riesch <michael.riesch@wolfvision.net>
---
 arch/arm64/boot/dts/rockchip/rk3568-evb1-v10.dts | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3568-evb1-v10.dts b/arch/arm64/boot/dts/rockchip/rk3568-evb1-v10.dts
index d8a4f7a9f562..39c495ff0157 100644
--- a/arch/arm64/boot/dts/rockchip/rk3568-evb1-v10.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3568-evb1-v10.dts
@@ -140,6 +140,11 @@ &gmac1m1_rgmii_clk
 	status = "okay";
 };
 
+&gpu {
+	mali-supply = <&vdd_gpu>;
+	status = "okay";
+};
+
 &i2c0 {
 	status = "okay";
 
@@ -462,6 +467,12 @@ &sdmmc0 {
 	status = "okay";
 };
 
+&tsadc {
+	rockchip,hw-tshut-mode = <1>;
+	rockchip,hw-tshut-polarity = <0>;
+	status = "okay";
+};
+
 &uart2 {
 	status = "okay";
 };
